PluginCore/PluginCore/Helpers/PathHelper.cs

```diff
@@ -19,10 +19,23 @@ public class PathHelper
         /// </summary>
         public static String BaseDir => PluginBase.MainForm.StandaloneMode ? AppDir : UserAppDir;
 
+        private static string appDir;
+
         /// <summary>
         /// Path to the main application directory
         /// </summary>
-        public static String AppDir => Path.GetDirectoryName(GetAssemblyPath(Assembly.GetExecutingAssembly()));
+        public static String AppDir
+        {
+            get
+            {
+                if (appDir == null)
+                {
+                    appDir = Path.GetDirectoryName(GetAssemblyPath(Assembly.GetExecutingAssembly()));
+                }
+
+                return appDir;
+            }
+        }
```

Comment:
Is it possible that the executing assembly may change at runtime or have several for the same app domain? I'm not sure, but even if it would, I think it makes sense to cache here.